src/bindings/python/flux/core/handle.py

```diff
@@ -38,7 +38,12 @@ def __init__(self, url=ffi.NULL, flags=0, handle=None):
         )
 
         if handle is None:
-            self.handle = raw.flux_open(url, flags)
+            try:
+                self.handle = raw.flux_open(url, flags)
+            except Exception as err:
+                raise EnvironmentError(
+                    err.errno, "Unable to connect to Flux: {}".format(err.strerror)
+                )
```

Comment:
Generic `Exception`s aren't guaranteed to have an `errno` field.  Only `EnvironmentError`s (and by proxy `IOError`s and `OSError`s) are documented to have one [(docs link)](https://docs.python.org/2/library/exceptions.html#exceptions.EnvironmentError).  So this exception handle can potentially cause it's own exception on the `err.errno` access.  I recommend limiting the scope of the exceptions caught from `Exception` to just `EnvironmentError`.

Reply:
Ok great! I really needed help with this code. For the life of me I couldn't find documentation on what functions throw which exceptions. Thanks!

Reply:
@SteVwonder is it ok to rethrow `EnvironmentError` in this case? Can you think of a better way to generally improve the error message from `flux.Flux()`?

Reply:
> @SteVwonder is it ok to rethrow EnvironmentError in this case? 

Yeah, I think that is totally reasonable.

> For the life of me I couldn't find documentation on what functions throw which exceptions. Thanks!

Yeah....that's definitely a big problem (in Python and C++).  Exception are great and yet still terrible.  Unfortunately, I don't have a better answer other than looking at the Python Docs (which can be a crap shot), and learning the hard way by shooting yourself in the foot :gun:.